### PR TITLE
erlcloud_application_autoscaler: fix error decoder by verifying if the response contains a json body

### PR DESCRIPTION
**Pull request by va17**
https://github.com/erlcloud/erlcloud/pull/662 introduced the error decoder but there is a bug in there where it assumes there will always be a JSON in the response body. This PR fix error decoder by verifying if the response contains a JSON body before attempting to decode it.
This PR also:
* Updates `ok_error_response` spec type accordingly
* Returns errors that are not decodable in the format of `erlcloud_aws:httpc_result_error()`
* Fix issue where errors are returned as `{error, {error, ...}}`

@nalundgaard

### CODE REVIEW

File: src/erlcloud_application_autoscaler.erl [hunk 1]
```diff
@@ -741,8 +741,13 @@ headers(Config, Operation, Body) ->
...
 %% Example: {error, {<<"ConcurrentUpdateException">>, <<"You already have a pending update to an Auto Scaling resource.">>}}
-decode_error(Response) ->
-    DecodedError = jsx:decode(Response#aws_request.response_body, [{return_maps, false}]),
-    ErrorType = proplists:get_value(<<"__type">>, DecodedError, <<>>),
-    ErrorMessage = proplists:get_value(<<"Message">>, DecodedError, <<>>),
-    {error, {ErrorType, ErrorMessage}}.
+decode_error(#aws_request{response_body = Body}) ->
+    case jsx:is_json(Body) of
+        false ->
+            {error, {<<>>, <<>>}};
```
Comment (nalundgaard) [inline src/erlcloud_application_autoscaler.erl:747]: Here can you actually return `{error, {http_error, Status, StatusLine, RespBody}}`? Should be able to get those all from the `#aws_request`.

Comment (va17) [reply src/erlcloud_application_autoscaler.erl]: Updated the PR to return the following:
```erlang
{error, {Response#aws_request.httpc_error_reason, Response#aws_request.response_status, Response#aws_request.response_status_line, Body}};
```

File: src/erlcloud_application_autoscaler.erl [hunk 2]
```diff
@@ -61,10 +61,14 @@
...
 -type response_key() :: atom().
 -type response() :: [{response_key(), response_attribute()}].
+-type error_type() :: binary().
+-type error_message() :: binary().
 -type ok_error_response() :: {ok, jsx:json_term()}
                            | {error, metadata_not_available
                                    | container_credentials_unavailable
-                                   | erlcloud_aws:httpc_result_error()}.
+                                   | erlcloud_aws:httpc_result_error()}
+                           | {error, {error_type(), error_message()}}
```
Comment (nalundgaard) [inline src/erlcloud_application_autoscaler.erl:70]: Please use the pattern here, not this redundant thing.

Comment (va17) [reply src/erlcloud_application_autoscaler.erl]: Fixed.

**Review by nalundgaard — APPROVED** [review]

**Pull request merged**